Lib/test/test_complex.py

```diff
@@ -99,20 +99,48 @@ def test_truediv(self):
             self.check_div(complex(random(), random()),
                            complex(random(), random()))
 
-        self.assertRaises(ZeroDivisionError, complex.__truediv__, 1+1j, 0+0j)
-        self.assertRaises(OverflowError, pow, 1e200+1j, 1e200+1j)
-
         self.assertAlmostEqual(complex.__truediv__(2+0j, 1+1j), 1-1j)
-        self.assertRaises(ZeroDivisionError, complex.__truediv__, 1+1j, 0+0j)
 
         for denom_real, denom_imag in [(0, NAN), (NAN, 0), (NAN, NAN)]:
             z = complex(0, 0) / complex(denom_real, denom_imag)
             self.assertTrue(isnan(z.real))
             self.assertTrue(isnan(z.imag))
 
+    def test_truediv_zero_division(self):
+        with self.assertRaises(ZeroDivisionError):
+            (1+1j) / (0+0j)
+        with self.assertRaises(ZeroDivisionError):
+            (1+1j) / 0.0
+        with self.assertRaises(ZeroDivisionError):
+            (1+1j) / 0
+        with self.assertRaises(ZeroDivisionError):
+            1.0 / (0+0j)
+        with self.assertRaises(ZeroDivisionError):
+            1 / (0+0j)
```

Comment:
It seems like these 5 tests are copy/pasted in multiple test methods. Would it be possible to use a global list and write a test using this test? Something like:

```
for a, b in ZERO_DIVISION:
    with self.subTest(a=a, b=b):
        with self.assertRaises(ZeroDivisionError):
            a / b
```

Reply:
Done.